src/generator/operations.ts

```diff
@@ -264,20 +264,32 @@ function generateOperation(op: Operation, imports: ImportManager): string {
   text += `\tif err != nil {\n`;
   text += `\t\treturn nil, err\n`;
   text += `\t}\n`;
-  text += `\tif !resp.HasStatusCode(${formatStatusCodes(statusCodes)}) {\n`;
-  text += `\t\treturn nil, client.${info.protocolNaming.errorMethod}(resp)\n`;
-  text += '\t}\n';
-  if (isLROOperation(op)) {
-    text += '\t return resp, nil\n';
-  } else if (needsResponseHandler(op)) {
-    // also cheating here as at present the only param to the responder is an azcore.Response
-    text += `\tresult, err := client.${info.protocolNaming.responseMethod}(resp)\n`;
-    text += `\tif err != nil {\n`;
-    text += `\t\treturn nil, err\n`;
-    text += `\t}\n`;
-    text += `\treturn result, nil\n`;
+  // HAB with schema response is handled in protocol responder
+  if (op.language.go!.headAsBoolean && !(op.responses && isSchemaResponse(op.responses[0]))) {
```

Comment:
This case will always be for operations that dont have responses? Or for operations that dont have schema responses? Just checking, we dont need to include it for those ever?

Reply:
It will be for the case where HAB is enabled and there's no schema response.

This is a corner case.  By definition, an HTTP HEAD request will never return a response body (i.e. it's never a schema response); just an HTTP status code and possibly header values (modeling a response body for HEAD is an error).  For operation responses that don't model a response body but *do* model response headers, we create a schema response modeling the header values.  So we have this case where a HEAD operation becomes a schema response to model the response headers.

If an operation models only response headers, it will return a `<Group><Method>Response` response envelope.  In the case of a HEAD request with HAB enabled, we must use the operation-specific response envelope, not the generic `BooleanResponse` response envelope.

Reply:
Got it, thank you for the explanation! :)